CoreFoundation/Base.subproj/CFInternal.h

```diff
@@ -171,9 +171,13 @@ CF_PRIVATE CFIndex __CFActiveProcessorCount();
 #define __builtin_unreachable() do { } while (0)
 #endif
 
-#if defined(__i386__) || defined(__x86_64__)
-    #if defined(__GNUC__)
-        #define HALT do {asm __volatile__("int3"); kill(getpid(), 9); __builtin_unreachable(); } while (0)
+#if defined(__i386__) || defined(__x86_64__) || defined(__s390x__)
```

Comment:
i386 and x86_64 are not defined on s390x.

Perhaps instead of doing a check on s390x, we can change the code to check like this?

```
#if defined(__GNUC__) 
        #if defined(__i386__) || defined(__x86_64__)
          #define HALT do {asm __volatile__("int3"); kill(getpid(), 9); __builtin_unreachable(); } while (0)
        #else
          #define HALT do {__builtin_trap(); kill(getpid(), 9); __builtin_unreachable(); } while (0)
        #endif
    #elif defined(_MSC_VER)
        #define HALT do { DebugBreak(); abort(); __builtin_unreachable(); } while (0)
    #else
        #error Compiler not supported
    #endif
```